pontoon/actionlog/migrations/0004_auto_20240816_1045.py

```diff
@@ -0,0 +1,48 @@
+# Generated by Django 4.2.11 on 2024-08-16 10:45
+
+import datetime
+
+from django.db import migrations
+
+
+def migrate_translation_to_actionlog(apps, schema_editor):
+    Translation = apps.get_model("base", "Translation")
+    ActionLog = apps.get_model("actionlog", "ActionLog")
+
+    traslation_info = {
+        "date": ActionLog.ActionType.TRANSLATION_CREATED,
+        "approved_date": ActionLog.ActionType.TRANSLATION_APPROVED,
+        "unapproved_date": ActionLog.ActionType.TRANSLATION_UNAPPROVED,
+        "rejected_date": ActionLog.ActionType.TRANSLATION_REJECTED,
+        "unrejected_date": ActionLog.ActionType.TRANSLATION_UNREJECTED,
+    }
+
+    # date to end migration
+    end_date = datetime.datetime(
+        2020, 1, 7, 9, 25, 11, 829125, tzinfo=datetime.timezone.utc
+    )
+
+    actions_to_log = []
+
+    for translation in Translation.objects.filter(date__lt=end_date):
+        for attr, action_type in traslation_info.items():
+            if getattr(translation, attr) is not None:
```

Comment:
We should also check that any date is before `end_date` in order to avoid duplicate entries. It could be that translation was created before we started using ActionLog, but it could have been approved afterwards.

Reply:
researched on this and saw that `django.db.models.Q` is a good way to do this
You'll see it in my latest update
Let me know what you think about it

Reply:
If the `Translation.date` is before `end_date` and `Translation.approved_date` isn't, we'll still create the (duplicate) `ActionLog` item.

It should be sufficient to check `Translation.date` only in the QuerySet (as we did before) and then make sure that `getattr(translation, attr)` is before `end_date`.